doc/testing.md

```diff
@@ -0,0 +1,186 @@
+# Testing Demikernel
+
+This document contains instructions on how to test Demikernel. What it is covered:
+
+- [What are unit tests](#what-are-unit-tests)
+- [How to run unit tests](#how-to-run-unit-tests)
+- [What are system-level tests](#what-are-system-level-tests)
+- [How to run system-level tests](#how-to-run-system-level-tests)
+
+> The instructions in this file assume that you have at your disposal at least one machine with Demikernel's development
+environment set up. For more information on how to set up Demikernel's development environment, check out instructions
+in the `README.md` file.
+
+## What Are Unit Tests
+
+Unit tests exercise internal components of Demikernel, individually.
+
+Demikernel has several unit tests, and all of them spread across the source tree in `demikernel/src/`. In case you want
+to check for their implementation, then search for `mod tests` and `cfg(test)` in the code base.
```

Comment:
I'd change this to:

Demikernel has several unit tests spread across the source tree in `demikernel/src/`.  To check their implementation, search for `mod tests` and `cfg(test)` in the code base.

Reply:
Done.